### ISSUE TITLE
npm shell <nodenum> does not work

### ISSUE THREAD

**Issue by hassan-xy** [project-chip/matter.js#898]
Hi team,
I am using ubuntu 20.4, kernel 4.9.56, node version v16.20.2, npm version 8.19.4. I installed the command  
npm install @project-chip/matter-node-shell.js"'.

I am trying to run using the "npm shell <nodenum>" but I get the error message as shown below.
![image](https://github.com/project-chip/matter.js/assets/125990448/d72b85c5-82d4-4166-8051-b0c66afcfd6e)

I tried to resolve the issue but somehow I am unable to. Please guide me.

**Comment by Apollon77** [comment]
"npm run shell 1" should be your friend

**Comment by hassan-xy** [comment]
@Apollon77 
![image](https://github.com/project-chip/matter.js/assets/125990448/c25a47c7-72da-4807-a5f5-5e383c7b9405)
Nope does not work. Maybe you can share a screenshot of what happens on your end and then I can proceed.

**Comment by Apollon77** [comment]
in which directory do you execute that? you might need to change into node_modules/@project-chip/matter-node-shell.js director for that to work? Seems we might need to adjust docs ...

**Comment by hassan-xy** [comment]
![image](https://github.com/project-chip/matter.js/assets/125990448/c7926654-fa52-4ed6-a2db-ccf3f69f656c)
So i tried it and I get a new error.

**Comment by hassan-xy** [comment]
Come one guys, give me something here.....

**Comment by Apollon77** [comment]
I did not had time to repro this, but in fact what you can alöways do is to call the scrips directly like

"node node_modules/@project-chip/matter-node-shell.js/dist/cjs/app.js ..."

**Comment by lauckhart** [comment]
@hassan-xy I addressed the bug you turned up with npm run in the installed package in #933.  You'll have to wait until it gets into the nightly to use it.  You also should be able to do `npx matter-shell 1` once that PR lands to run the shell when the module is installed.

@Apollon77's last answer should always work too -- just run the JS directly with `node`.

**Comment by Apollon77** [comment]
Ok, fixes are in latest nightlies

**Comment by hassan-xy** [comment]
![image](https://github.com/project-chip/matter.js/assets/125990448/bc0e6e15-118d-4704-818e-51cb7e007e0f)
Not sure whats happening here but just cant seem to detect nearby devices. I have a esp32 that gets detected by the chip tool but does not detect here.

**Comment by hassan-xy** [comment]
Guys any updates?

**Comment by Apollon77** [comment]
So great to see it runs now. In fact this message mainly means that he was not able to discover any devoice as you correctly assume. for details please set the loglevel of the shell a bit higher or enable debug logfile (see "config help" for options) and porovide this - but pleae in a new issue because the isue here is closed because it launches now
